docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -1073,6 +1073,51 @@ When you plan to use bearer access tokens during the opening WebSocket handshake
 * Use a custom WebSocket ticket system which supplies a random token with the HTML page which hosts the JavaScript WebSockets client which must provide this token during the initial handshake request as a query parameter.
 ====
 
+Before the bearer access token sent on the initial HTTP request expires, you can send a new bearer access token as part of a message and update current `SecurityIdentity` attached to the WebSocket server connection:
+
+[source, java]
+----
+package io.quarkus.websockets.next.test.security;
+
+import io.quarkus.security.Authenticated;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.websockets.next.OnTextMessage;
+import io.quarkus.websockets.next.WebSocket;
+import io.quarkus.websockets.next.WebSocketSecurity;
+import jakarta.inject.Inject;
+
+@Authenticated
+@WebSocket(path = "/end")
+public class Endpoint {
+
+    record RequestDto(String accessToken, String message) {}
+
+    @Inject
+    SecurityIdentity securityIdentity;
+
+    @Inject
+    WebSocketSecurity webSocketSecurity;
+
+    @OnTextMessage
+    String echo(RequestDto request) {
```

Comment:
@michalvavrik 

I think this method will have to match the security level of the HTTP upgrade.

For example, if HTTP Upgrade happened at `@RolesAllowed("Admin")`, then only verifying the validity of the token  which would only match `@Authenticated` level, would raise a risk of decreasing the original security identity strength, for example, the new token which is about to replace an `admin`  level identity may only have a `user` role.

Reply:
OK, that's easy to do.

Reply:
@michalvavrik Sounds good,  we do it indirectly in the code flow too  when refreshing tokens, since it is done in scope of the request.
In case of this PR, the new token is coming via the front channel so the extra care is needed. 
We might end up doing extra checks as well...

Reply:
I am not sure if I understand what you mean, but my plan is: there is a check that assures principal name is same (so they can't change alice for bob), but for roles and permissions, we just need to reapply standard security annotations (HTTP policies sounds like out of scope to me), which I _think is easy_. There is a code that prevents repeated security checks, I'll just disable the code when the identity refresh is enabled.

Reply:
@michalvavrik I was actually thinking about it too, how do you prevent an identity swap, so having the principal name check is good, but a few more checks may be necessary, such as a `sub` claim - `sub` is a unique number associated with a given user that did an original login, or even a more thorough identity comparison, but we can think about it later.

IMHO, whatever the security constraints were applied to HTTP upgrade should be reapplied at the token injection time because this is how we do in the code flow refresh. For example, when the ID and access token are refreshed, the refreshed ID token goes through the complete authentication and authorization cycle before the method cane be called.

Reply:
If the token is binary then we can update OidcIdentityProvider to record `sub` value from the introspection response, something like that, so that you can easily access this `sub`

Reply:
I have added `sub` check on the introspection result, but the documented QS test and the tests I wrote have `null` introspection attribute. Aren't:

1. check principal
3. re-apply security checks

good proves? Both tested.